From 336185998512551b6cef32412b7ec5cddb7b6df8 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Date: Fri, 3 Oct 2025 12:30:09 +0200
Subject: [PATCH 928/954] feat(st-spi): add fixed clock support in
 stm32_ospi_init

Use new clock API to consider fixed clock as source for the OSPI
driver. Keep compliance with nominal case.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git bcdcce7861941fd9fb1e7aaa342e7ed9427f2947 feat(st-spi): add fixed clock support in stm32_ospi_init]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Change-Id: I20f19406a9ab561159250eaeb7789aa4a4571abb
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/496149
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
---
 drivers/st/spi/stm32_ospi.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/st/spi/stm32_ospi.c b/drivers/st/spi/stm32_ospi.c
index 748ebf3597..458d210945 100644
--- a/drivers/st/spi/stm32_ospi.c
+++ b/drivers/st/spi/stm32_ospi.c
@@ -1353,13 +1353,10 @@ int stm32_ospi_init(void)
 				     mm_base : mm_base + mm_size - stm32_ospi.mm_size;
 		stm32_ospi.bank = bank;
 
-		cuint = fdt_getprop(fdt, ospi_node, "clocks", NULL);
-		if (cuint == NULL) {
-			return -FDT_ERR_BADVALUE;
+		ret = dt_get_clk_by_index(fdt, ospi_node, 0, &stm32_ospi.clock_id);
+		if (ret != 0) {
+			return ret;
 		}
-
-		cuint++;
-		stm32_ospi.clock_id = (unsigned long)fdt32_to_cpu(*cuint);
 	}
 
 	clk_enable(stm32_ospi.clock_id);
-- 
2.53.0

